### PR TITLE
Make initial run variable available to addressable_lambda

### PR DESCRIPTION
**Pull request by Skaronator**
## Description:

This PR adds a new variable, called initial_run, for the addressable_lambda. This variable is especially useful when the addressable lambda effect has some variables which need to be resetted with each restart.

Example

```yaml

effects:
  - addressable_lambda:
      name: Sunrise
      lambda: |-
        static uint16_t middlePos = it.size() / 2;
        static uint16_t ambientSize;
        static uint8_t ambientProgress;
        if (initial_run) {
          it.all() = ESPColor::BLACK;
          ambientSize = 0;
          ambientProgress = 20;
          return;
        }

        if (ambientProgress < 250) {
           ambientProgress++;
        }
        if (ambientSize < middlePos) {
           ambientSize++;
        }
        it.range(middlePos-ambientSize, middlePos+ambientSize) = ESPColor(ambientProgress, 0, 0, ambientProgress/4);
```

**Related issue (if applicable):** -

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#558

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/light/addressable_light_effect.h [hunk 1]
```diff
@@ -50,21 +50,27 @@ class AddressableLightEffect : public LightEffect {
...
                                uint32_t update_interval)
       : AddressableLightEffect(name), f_(f), update_interval_(update_interval) {}
+  void start() override { this->initial_run_ = true; }
   void apply(AddressableLight &it, const ESPColor &current_color) override {
     const uint32_t now = millis();
     if (now - this->last_run_ >= this->update_interval_) {
       this->last_run_ = now;
-      this->f_(it, current_color);
+      this->f_(it, current_color, this->initial_run_);
+      if (this->initial_run_) {
```
Comment (glmnet) [inline esphome/components/light/addressable_light_effect.h:63]: you can just set this to false

**Comment by glmnet** [comment]
LGTM Docs too

File: esphome/components/light/addressable_light_effect.h (hunk 1 above)
Comment (Skaronator) [reply esphome/components/light/addressable_light_effect.h]: Yup I've removed it

**Comment by glmnet** [comment]
Thanks @Skaronator 

I was wondering, it'd be nice to people to share somehow their effects

**Pull request merged**